include/app_memory/app_macro_support.h

```diff
@@ -0,0 +1,32 @@
+#ifndef _APP_MACRO_SUPPORT__H_
+#define _APP_MACRO_SUPPORT__H_
+
+/*
+ * The following provides variadic preprocessor macro support to
+ * help eliminate multiple, repetitive function/macro calls.  This
+ * allows up to 10 "arguments" in addition to _call .
+ * Note - derived from work on:
+ * https://codecraft.co/2014/11/25/variadic-macros-tricks/
+ */
+
+#define _GET_ARG(_1, _2, _3, _4, _5, _6, _7, _8, _9, _10, N, ...) N
+
+#define _for_0(_call, ...)
+#define _for_1(_call, x) _call(x)
+#define _for_2(_call, x, ...) _call(x) _for_1(_call, ##__VA_ARGS__)
+#define _for_3(_call, x, ...) _call(x) _for_2(_call, ##__VA_ARGS__)
+#define _for_4(_call, x, ...) _call(x) _for_3(_call, ##__VA_ARGS__)
+#define _for_5(_call, x, ...) _call(x) _for_4(_call, ##__VA_ARGS__)
+#define _for_6(_call, x, ...) _call(x) _for_5(_call, ##__VA_ARGS__)
+#define _for_7(_call, x, ...) _call(x) _for_6(_call, ##__VA_ARGS__)
+#define _for_8(_call, x, ...) _call(x) _for_7(_call, ##__VA_ARGS__)
+#define _for_9(_call, x, ...) _call(x) _for_8(_call, ##__VA_ARGS__)
+#define _for_10(_call, x, ...) _call(x) _for_9(_call, ##__VA_ARGS__)
+
+#define FOR_EACH(x, ...) \
+	_GET_ARG(__VA_ARGS__, \
+	_for_10, _for_9, _for_8, _for_7, _for_6, _for_5, \
+	_for_4, _for_3, _for_2, _for_1, _for_0)(x, ##__VA_ARGS__)
```

Comment:
This doesn't seem specific to app_memory, it would be unfortunate if a new component also required this and we had to either duplicate the macros or make references to app_memory.

Perhaps it would be better located closer to the other variadic preprocessor macros in zephyr/include/misc/util.h